src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1504,39 +1504,60 @@ def generate(
             generation_config = self.generation_config
 
         if return_timestamps is not None:
-            generation_config.return_timestamps = return_timestamps
-
-        if task is not None:
-            generation_config.task = task
+            if not hasattr(generation_config, "no_timestamps_token_id"):
+                raise ValueError(
+                    "You are trying to return timestamps, but the generation config is not properly set."
+                    "Make sure to initialize the generation config with the correct attributes that are needed such as `no_timestamps_token_id`."
+                    "For more details on how to generate the approtiate config, refer to https://github.com/huggingface/transformers/issues/21878#issuecomment-1451902363"
+                )
 
-        if is_multilingual is not None:
-            generation_config.is_multilingual = is_multilingual
+            generation_config.return_timestamps = return_timestamps
+        else:
+            generation_config.return_timestamps = False
 
         if language is not None:
             generation_config.language = language
+        if task is not None:
+            generation_config.task = task
 
         forced_decoder_ids = []
-
-        if hasattr(generation_config, "is_multilingual") and generation_config.is_multilingual:
+        if task is not None or language is not None:
             if hasattr(generation_config, "language"):
-                forced_decoder_ids.append((1, generation_config.lang_to_id[generation_config.language]))
+                if generation_config.language in generation_config.lang_to_id.keys():
+                    language_token = generation_config.language
+                elif generation_config.language in TO_LANGUAGE_CODE.keys():
+                    language_token = f"<|{TO_LANGUAGE_CODE[generation_config.language]}|>"
+                else:
+                    raise ValueError(f"The `{generation_config.language}` language token is not supported.")
```

Comment:
I get a lot of questions from people asking what format the language has to be in - should we update the docstring with the new language options and also give people the list of possible languages if they pass an invalid language? Along the lines of: https://github.com/huggingface/transformers/blob/a9bd5df16a46356463f2712dd8f6c109fa83d6f9/src/transformers/models/whisper/tokenization_whisper.py#L407-L408

Reply:
yep that's exact